docs/observability.md

```diff
@@ -0,0 +1,126 @@
+# COS Integration
+
+## (User) Getting Started
+
+This section describes how to deploy a single-node MicroK8s cluster with COS, and then relate MicroK8s with COS to get metrics, logs and dashboards for the cluster.
+
+### Deploy MicroK8s
+
+```bash
+# Add a machine model and deploy MicroK8s
+juju add-model microk8s
+juju deploy microk8s --channel edge --config hostpath_storage=true --constraints 'mem=8G'
+juju expose microk8s
+
+# Add MicroK8s as Kubernetes cloud to Juju
+juju run --unit microk8s/leader -- microk8s config | juju add-k8s k8s --controller "$(juju controller-config controller-name)"
+```
+
+### Deploy cos-lite
+
+```bash
+# Add a Kubernetes model and deploy MetalLB for LoadBalancer support
+juju add-model metallb-system k8s
+juju deploy metallb-controller --config iprange=10.43.5.10-10.43.5.20
+juju deploy metallb-speaker
+
+# Add a Kubernetes model and deploy cos-lite
+juju add-model observability k8s
+juju deploy cos-lite --trust --channel edge
+
+# Expose metrics, logs and dashboards relations
+juju offer prometheus:receive-remote-write prometheus
+juju offer loki:logging loki
+juju offer grafana:grafana-dashboard grafana
+```
+
+### Relate MicroK8s with COS
+
+```bash
+# Switch to 'microk8s' model and configure cross-model relations
+juju switch microk8s
+juju consume admin/observability.prometheus prometheus
+juju consume admin/observability.loki loki
+juju consume admin/observability.grafana grafana
+
+# Deploy grafana-agent and integrate grafana-agent with COS
+juju deploy grafana-agent --channel edge
+juju relate grafana-agent prometheus
+juju relate grafana-agent loki
+juju relate grafana-agent grafana
+
+# Integrate MicroK8s with grafana-agent
+juju relate microk8s grafana-agent
+```
+
+### Access Grafana
+
+Retrieve the admin password for Grafana using:
+
+```bash
+juju run-action --wait -m observability grafana/0 get-admin-password
+```
+
+Then, if LoadBalancer IP is not reachable from your host, use either `sshuttle` or `ssh`
+
+```bash
+# Access Grafana at http://localhost:8000/observability-grafana/
+juju ssh -m microk8s microk8s/leader -L 8000:10.43.5.10:80
```

Comment:
Should we reduce the range of IPs metallb has?

Reply:
This just shows an example range with internal IPs. It is up to the user to change this according to the deployment requirements.